installer/src/main/java/com/teamscale/profiler/installer/TeamscaleUtils.java

```diff
@@ -0,0 +1,104 @@
+package com.teamscale.profiler.installer;
+
+import okhttp3.Credentials;
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+import javax.net.ssl.SSLException;
+import java.io.IOException;
+import java.net.ConnectException;
+import java.net.SocketTimeoutException;
+import java.net.UnknownHostException;
+
+/** Utilities for interacting with a Teamscale instance. */
+public class TeamscaleUtils {
+
+	/** Ensures that we can connect to Teamscale with the given credentials. */
+	public static void checkTeamscaleConnection(TeamscaleCredentials credentials) throws FatalInstallerError {
+		OkHttpClient client = new OkHttpClient();
+
+		// we use the project list as a test to see if the Teamscale credentials are valid.
+		// any other API that requires a logged-in user would be fine as well.
+		HttpUrl url = credentials.url.resolve("api/v8.8/projects");
+		if (url == null) {
+			// this should never happen but the API forces us to handle this
+			throw new IllegalStateException("Cannot resolve API endpoint against URL " + credentials.url);
+		}
+
+		Request request = new Request.Builder()
+				.url(url)
+				.header("Authorization", Credentials.basic(credentials.username, credentials.accessKey))
+				.build();
+
+		try (Response response = client.newCall(request).execute()) {
+			handleErrors(response, credentials);
+		} catch (SSLException e) {
+			throw new FatalInstallerError("Failed to connect via HTTPS to " + credentials.url
+					+ "\nPlease ensure that your Teamscale instance is reachable under " + credentials.url
+					+ " and that it is configured for HTTPS, not HTTP. E.g. open that URL in your"
+					+ " browser and verify that you can connect successfully.",
```

Comment:
Might also happen if the server uses an internally signed SSL certificate that is not installed on the current machine, right?